pkg/initiailizer_v2/model/huggingface.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from urllib.parse import urlparse
+
+import huggingface_hub
+
+import pkg.initiailizer_v2.utils.utils as utils
+
+# TODO (andreyvelich): This should be moved to SDK V2 constants.
+import sdk.python.kubeflow.storage_initializer.constants as constants
+from pkg.initiailizer_v2.model.config import HuggingFaceModelInputConfig
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] %(message)s",
+    datefmt="%Y-%m-%dT%H:%M:%SZ",
+    level=logging.INFO,
+)
+
+
+class HuggingFace(utils.ModelProvider):
+
+    def load_config(self):
+        config_dict = utils.get_config_from_env(HuggingFaceModelInputConfig)
+        logging.info(f"Config for HuggingFace model initiailizer: {config_dict}")
+        self.config = HuggingFaceModelInputConfig(**config_dict)
+
+    def download_model(self):
+        storage_uri_parsed = urlparse(self.config.storage_uri)
+        model_uri = storage_uri_parsed.netloc + storage_uri_parsed.path
+
+        logging.info(f"Downloading model: {model_uri}")
+        logging.info("-" * 40)
+
+        if self.config.access_token:
+            huggingface_hub.login(self.config.access_token)
+
+        # TODO (andreyvelich): We should verify these patterns for different models.
+        huggingface_hub.snapshot_download(
+            repo_id=model_uri,
+            local_dir=constants.VOLUME_PATH_MODEL,
+            allow_patterns=["*.json", "*.safetensors", "*.model"],
```

Comment:
For `"*.safetensors"` and `"*.model"` in the allow_patterns, I would say it works in most of the cases.
However, for model like [mistralai/Mistral-7B-Instruct-v0.3](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3/tree/main), it has [consolidated.safetensors](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3/blob/main/consolidated.safetensors) and [tokenizer.model.v3](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3/blob/main/tokenizer.model.v3) (so called v3 format in Mistral, check their download example [here](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3#download)).
```python
snapshot_download(repo_id="mistralai/Mistral-7B-Instruct-v0.3", allow_patterns=["params.json", "consolidated.safetensors", "tokenizer.model.v3"], local_dir=mistral_models_path)
```
In this case, downloading the above mistral model with the current allow_patterns, the downloaded size will be 29 GB (double from the actual size 14.5 GB). Probably you need some logic to handle the Mistral model.

Reply:
Thanks for sharing! Let me add it to the TODO list.